### PR TITLE
DOC-737: Models

### PR DESCRIPTION
**Pull request by elenaili**

### CODE REVIEW

**Review by phearnot — APPROVED** [review]

File: node/src/main/resources/swagger-ui/swagger.json [hunk 1]
```diff
@@ -4250,674 +4980,2182 @@
...
+          "key": {
+            "type": "string"
+          },
+          "type": {
+            "type": "string",
+            "enum": ["integer"]
+          },
+          "value": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5483]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 2]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "proofs": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          },
+          "matcherFeeAssetId": {
+            "type": "string",
+            "format": "nullable"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5588]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 3]
```diff
@@ -4250,674 +4980,2182 @@
...
+          "senderPublicKey": {
+            "type": "string"
+          },
+          "fee": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "feeAssetId": {
+            "type": "string",
+            "format": "nullable"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5909]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 4]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5980]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 5]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6047]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 6]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6138]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 7]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6394]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 8]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6482]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 9]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6553]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 10]
```diff
@@ -4250,674 +4980,2182 @@
...
+          "senderPublicKey": {
+                "type": "string"
+          },
+          "fee": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "feeAssetId": {
+            "type": "string",
+            "format": "nullable"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6631]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 11]
```diff
@@ -4250,674 +4980,2182 @@
...
+          "senderPublicKey": {
+            "type": "string"
+          },
+          "fee": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "feeAssetId": {
+            "type": "string",
+            "format": "nullable"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6727]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 12]
```diff
@@ -4250,674 +4980,2182 @@
...
+          "senderPublicKey": {
+                "type": "string"
+          },
+          "fee": {
+             "type": "integer",
+             "format": "int32"
+          },
+          "feeAssetId": {
+            "type": "string",
+            "format": "nullable"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6803]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 13]
```diff
@@ -4250,674 +4980,2182 @@
...
+          "senderPublicKey": {
+            "type": "string"
+          },
+          "fee": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "feeAssetId": {
+            "type": "string",
+            "format": "nullable"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6875]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 14]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6944]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 15]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:7018]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 16]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "recipient": {
+            "type": "string"
+          },
+          "assetId": {
+            "type": "string"
+          },
+          "feeAsset": {
+            "type": "string",
+            "format": "nullable"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:7046]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 17]
```diff
@@ -4250,674 +4980,2182 @@
...
+          "senderPublicKey": {
+            "type": "string"
+          },
+          "fee": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "feeAssetId": {
+            "type": "string",
+            "format": "nullable"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:7102]: "nullable": true

File: node/src/main/resources/swagger-ui/swagger.json [hunk 18]
```diff
@@ -4250,674 +4980,2182 @@
...
+                "nullable": true
+              }
+            }
+          },
+          "orderType": {
+            "type": "string"
+          },
+          "amount": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5559]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 19]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "price": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5563]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 20]
```diff
@@ -4250,674 +4980,2182 @@
...
+            "type": "integer",
+            "format": "int64"
+          },
+          "expiration": {
+            "type": "integer",
+            "format": "int64"
+          },
+          "matcherFee": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5575]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 21]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "sender": {
+            "type": "string"
+          },
+          "senderPublicKey": {
+            "type": "string"
+          },
+          "fee": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6390]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 22]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "height": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "totalFee": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5171]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 23]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5263]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 24]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "sender": {
+            "type": "string"
+          },
+          "senderPublicKey": {
+            "type": "string"
+          },
+          "fee": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5905]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 25]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5976]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 26]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6043]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 27]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "order1": {
+            "$ref": "#/components/schemas/Order"
+          },
+          "order2": {
+            "$ref": "#/components/schemas/Order"
+          },
+          "amount": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6162]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 28]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "price": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6166]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 29]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "buyMatcherFee": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6170]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 30]
```diff
@@ -4250,674 +4980,2182 @@
...
+            "type": "string"
+          },
+          "sender": {
+            "type": "string"
+          },
+          "senderPublicKey": {
+            "type": "string"
+          },
+          "fee": {
+            "type": "number"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6205]: integer, int64?

File: node/src/main/resources/swagger-ui/swagger.json [hunk 31]
```diff
@@ -4250,674 +4980,2182 @@
...
+            "type": "integer",
+            "format": "int32"
+          },
+          "buyMatcherFee": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "sellMatcherFee": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6174]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 32]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "sender": {
+            "type": "string"
+          },
+          "senderPublicKey": {
+            "type": "string"
+          },
+          "fee": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6478]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 33]
```diff
@@ -4250,674 +4980,2182 @@
...
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6549]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 34]
```diff
@@ -4250,674 +4980,2182 @@
...
+          },
+          "chainId": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "amount": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6575]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 35]
```diff
@@ -4250,674 +4980,2182 @@
...
+          "version": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "assetId": {
+            "type": "string"
+          },
+          "quantity": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:6748]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 36]
```diff
@@ -4250,674 +4980,2182 @@
...
+        ],
+        "type": "object",
+        "properties": {
+          "key": {
+            "type": "string"
+          },
+          "value": {
+            "type": "object",
+            "properties": {},
+            "description": "null"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5462]: null without "

**Review by DrBlast — CHANGES_REQUESTED** [review]
check all fees and amounts - format should be  int64

File: node/src/main/resources/swagger-ui/swagger.json (hunk 36 above)
Comment (elenaili) [reply node/src/main/resources/swagger-ui/swagger.json]: "null" here is a comment (I've added a more detailed explanation now)

File: node/src/main/resources/swagger-ui/swagger.json [hunk 37]
```diff
@@ -3427,116 +3957,130 @@
...
+                      "required": [
+                        "currentBestChannel",
+                        "localScore",
+                        "scoresCacheSize"
+                      ],
+                      "type": "object",
+                      "properties": {
+                        "localScore": {
+                          "type": "integer",
+                          "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:4047]: int64

File: node/src/main/resources/swagger-ui/swagger.json [hunk 38]
```diff
@@ -4250,674 +4980,2189 @@
...
+        ],
+        "type": "object",
+        "properties": {
+          "height": {
+            "type": "integer",
+            "format": "int32"
+          },
+          "totalWavesAmount": {
+            "type": "integer",
+            "format": "int32"
```
Comment (DrBlast) [inline node/src/main/resources/swagger-ui/swagger.json:5615]: int64

**Review by DrBlast — CHANGES_REQUESTED** [review]

**Review by DrBlast — APPROVED** [review]

**Pull request merged**